jQuery/ajax.js

```diff
@@ -1,82 +1,113 @@
+      { field: 'size', value: size.val() }
+    ];
+  }
+
+  function redrawContent(contentSelector, citizens) {
+    var content = $(contentSelector);
+    content.empty();
+
+    citizens.forEach(function(citizen) {
+      content.append($('<div>', { class: "citizen", html: citizen.name }));
+    });
+  }
+
+  return {
+    gatherParams: gatherParams,
+    populateSelectWithOptions: populateSelectWithOptions,
+    redrawContent: redrawContent
   };
+})();
+
+/* Defines funtions that handle backend logic that handle data, nothing to do with DOM */
+window.hh.logic = window.hh.logic || (function() {
+  function getUniqueValuesForField(field, citizens) {
+    var values = ['All'];
+    citizens.forEach(function(c) {
+      var v = c[field];
+      if (values.indexOf(v) === -1) {
+        values.push(v);
+      }
+    });
+    return values;
+  }
+
+  function filter(params, citizens) {
+    return citizens.filter(function(citizen) {
+      var matchesFilter = true;
+
+      for (var i = 0; i < params.length && matchesFilter; i++) {
+        var param = params[i];
+
+        var field = param['field'];
+        var citizenValue = citizen[field];
+        var paramValue = param['value'];
+
+        matchesFilter = paramValue === 'All' || citizenValue.toUpperCase().indexOf(paramValue.toUpperCase()) !== -1;
+      }
+      return matchesFilter;
+    });
+  }
+
+  return {
+    getUniqueValuesForField: getUniqueValuesForField,
+    filter: filter
+  };
+})();
 
-  $('#character-select').on('change', { citizens: citizens }, selectHandler);
-  $('#hometown-select').on('change', { citizens: citizens }, selectHandler);
-  $('#size-select').on('change', { citizens: citizens }, selectHandler);
-
-  selectHandler({data: { citizens: citizens }});
-}
-
-function getUniqueValuesForField(field, citizens) {
-  var values = ['All'];
-  citizens.forEach(function(c) {
-    var v = c[field];
-    if (values.indexOf(v) === -1) {
-      values.push(v);
-    }
-  });
-  return values;
-}
-
-function populateSelectWithOptions(select, options) {
-  options.forEach(function(option) {
-    $(select).append($('<option>', { value: option, html: option }));
-  });
-}
-
-function gatherParams() {
-  var params = [];
-
-  var characterValue = $('#character-select').val();
-  params.push({field: 'character', value: characterValue});
-
-  var hometownValue = $('#hometown-select').val();
-  params.push({field: 'hometown', value: hometownValue});
-
-  var sizeValue = $('#size-select').val();
-  params.push({field: 'size', value: sizeValue});
-
-  return params;
-}
-
-function filter(params, citizens) {
-  var filteredCitizens = citizens.filter(function(citizen) {
-    var matchesFilter = true;
-
-    for (var i = 0; i < params.length && matchesFilter; i++) {
-      var param = params[i];
-
-      var field = param['field'];
-      var citizenValue = citizen[field];
-      var paramValue = param['value'];
+window.hh.ajax = window.hh.ajax || function(citizens) {
```

Comment:
ah, I see, can't really do dependency injection as cleanly this way. I'll refactor